geps/gep-2627/index.md

```diff
@@ -0,0 +1,50 @@
+# GEP-2627: DNS configuration within Gateway API
+
+* Issue: [#2627](https://github.com/kubernetes-sigs/gateway-api/issues/2627)
+* Status: Provisional
+
+## TLDR
+
+For gateway infrastructure to be valuable we need to be able to connect clients to these gateways. A common way to achieve this is to use domain names/hostnames and DNS. Gateways define listeners that can have assigned hostnames or wildcards.  The guidelines for DNS configuration are a critical piece of service networking, but this is currently not expressible as part of Gateway API.   Instead of leaving this as an exercise for the user to figure out, this proposal attempts to provide options to ease Gateway API operations.
+
+## Goals
+* Provide a way for Gateway API resource owners to mark their resources as relevant for external DNS provisioning
+* Ensure that the above method has a way for multiple providers to be present in the cluster and be able to actuate external DNS provisioning requests
+* Ensure that any method is based on structured fields and makes the most of `status` on whatever resources are relevant, whether they are existing Gateway API resources or new resources.
+* Increase portability and supportability between Gateway API implementations and third party controllers offering DNS integration.
```

Comment:
To me this suggests to me some desire to go beyond just adding specification and status for which users can request DNS configuration for a `Gateway`, and the implementation can go deploy that and report status. This seems to suggest some greater involvement in the actual provisioning tooling, is that what was intended?

Reply:
The intention is to suggest that with this API I as the user can choose from any provider that supports this API similar to how Gateway API itself works with the `controllerName` in the GatewayClass. So if this API was implemented by `controller A` and I wanted to move to a new implementation in controller B that would be possible and not need a different API to achieve that